libs/clients/disability-license/src/lib/disabilityLicenseClient.provider.ts

```diff
@@ -8,16 +8,17 @@ import {
 } from '@island.is/nest/config'
 import { DisabilityLicenseClientConfig } from './disabilityLicenseClient.config'
 import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import { DisabilityLicenseService } from './disabilityLicenseClient.service'
 
-export const DisabilityLicenseApiProvider: Provider<DefaultApi> = {
-  provide: DefaultApi,
+export const DisabilityLicenseApiProvider: Provider<DisabilityLicenseService> = {
+  provide: DisabilityLicenseService,
```

Comment:
I'd suggest reverting changes to this file and let NestJS pass DefaultApi to the DisabilityLicenseService constructor.

You can stop exporting DefaultApi and DisabilityLicenseApiProvider from the client library. The client module can provide both DisabilityLicenseApiProvider and DisabilityLicenseService but only export DisabilityLicenseService.

So in other modules you import the DisabilityLicenseModule, and then get DisabilityLicenseService injected where its needed.

Reply:
yeah, that makes more sense